core/schema/sdk.go

```diff
@@ -763,6 +858,46 @@ func (sdk *goSDK) baseWithCodegen(
 		},
 	}
 
+	var config goSDKConfig
+	err = mapstructure.Decode(sdk.rawConfig, &config)
```

Comment:
note: unknown fields should cause an error like for `moduleSDK`. So setting `foobar` as a `config` field should error out here.